packages/pyright-internal/src/analyzer/checker.ts

```diff
@@ -256,6 +255,21 @@ export class Checker extends ParseTreeWalker {
     }
 
     check() {
+        // This checker does 2 main works.
+        // First is calling type eval on the appropriate nodes so that we can force
+        // type evaluation to happen and generate diagnostics if there are issues.
+        // Since we are lazy evaluator, unless someone calls type eval, no diagnostic
+        // will be generated. This walker basically walks entire tree and force evaluation.
+        // Second is using the result of the evaluation, look for issues that require larger
+        // context such as whether overloads have consistent signature. or diagnostics that
+        // doesn't require internal state of type eval to generate.
+
+        // This walker bypasses visiting ModuleNode, so checking module level complexity
+        // directly here.
+        if (!this._checkModuleComplexity()) {
```

Comment:
I am not sure why the walker bypass visitModule and have check method that directly calls statements and etc. this all seems could be done in visitModule method.